### PR TITLE
Support DuckDB in `rows_insert()`, `rows_update()` and `rows_delete()`

### PR DESCRIPTION
**Pull request by mgirlich**
Supporting DuckDB in these three functions seems to be pretty easy. Unfortunately, DuckDB doesn't seem to support RETURNING. At least I didn't find in the [documentation](https://duckdb.org/docs/sql/statements/insert) and got nothing back when I tried it out at first.

I guess that we could also remove `sql_rows_update.tbl_SQLiteConnection()` and let SQLite simply use `sql_rows_update.tbl_sql()`. What do you think?

### CODE REVIEW

File: R/rows-db.R [hunk 1]
```diff
@@ -511,6 +534,13 @@ sql_returning_cols.tbl_dbi <- function(x, returning_cols, ...) {
   paste0("RETURNING ", returning_cols)
 }
 
+#' @export
+sql_returning_cols.tbl_duckdb_connection <- function(x, returning_cols, ...) {
+  if (!is_empty(returning_cols)) {
+    abort("DuckDB does not support the `returning` argument.")
+  }
```
Comment (krlmlr) [inline R/rows-db.R:541]: Emptiness is handled by the generic already?

```suggestion
  abort("DuckDB does not support the `returning` argument.")
```

File: R/rows-db.R [hunk 2]
```diff
@@ -423,13 +445,14 @@ sql_rows_delete.tbl_sql <- function(x, y, by, ..., returning_cols = NULL) {
...
-    "WITH ", p$y_name, "(", p$y_columns_qq, ") AS (\n",
-    dbplyr::sql_render(y),
-    "\n)\n",
-    #
     "DELETE FROM ", p$name, "\n",
     sql_output_cols(x, returning_cols, delete = TRUE),
-    "WHERE EXISTS (SELECT * FROM ", p$y_name, " WHERE ", p$compare_qual_qq, ")",
+    "WHERE EXISTS (\n",
+    "  SELECT * FROM (\n",
+    "    ", dbplyr::sql_render(y), "\n",
```
Comment (krlmlr) [inline R/rows-db.R:452]: What's the motivation for this change? How/why did it work before?

**Review by krlmlr — COMMENTED** [review]
Thanks! Yeah, the fewer specialized methods we have the better.

**Comment by krlmlr** [comment]
Is there an existing DuckDB issue regarding support for `RETURNING` ?

CC @hannesmuehleisen.

File: R/rows-db.R (hunk 2 above)
Comment (mgirlich) [reply R/rows-db.R:452]: This just replaces the CTE with a subquery to make it work with DuckDB (CTEs don't work with INSERT, UPDATE, DELETE...)

**Review by krlmlr — APPROVED** [review]
Thanks!

**Comment by mgirlich** [comment]
> Is there an existing DuckDB issue regarding support for `RETURNING` ?
> 
> CC @hannesmuehleisen.

I didn't find an issue yet and therefore created one: https://github.com/duckdb/duckdb/issues/2291.
